pkg/xds/ingress/dataplane.go

```diff
@@ -162,11 +164,24 @@ func getIngressAvailableMeshGateways(meshName string, meshGateways []*core_mesh.
 	return tagSets.toAvailableServices()
 }
 
-func GetIngressAvailableServices(others []*core_mesh.DataplaneResource) []*mesh_proto.ZoneIngress_AvailableService {
+func GetIngressAvailableServices(others []*core_mesh.DataplaneResource, tagFilters []string) []*mesh_proto.ZoneIngress_AvailableService {
 	tagSets := tagSets{}
 	for _, dp := range others {
 		for _, dpInbound := range dp.Spec.GetNetworking().GetHealthyInbounds() {
-			tagSets.addInstanceOfTags(dp.GetMeta().GetMesh(), dpInbound.Tags)
+			tags := map[string]string{}
+			for key, value := range dpInbound.Tags {
+				matchAnyFilter := len(tagFilters) == 0
+				for _, tagFilter := range tagFilters {
+					if strings.HasPrefix(key, tagFilter) {
+						matchAnyFilter = true
+						break
+					}
+				}
+				if matchAnyFilter {
+					tags[key] = value
+				}
+			}
```

Comment:
There are 4 nested for loops now, which makes it hard to read and understand. Could we abstract it to separate function `filterTags`?

As `slices` package will be added to stdlib, maybe we could simplify this logic a little bit by using [`slices.ContainsFunc`](https://pkg.go.dev/golang.org/x/exp/slices#ContainsFunc)?
Something like:
```suggestion
			tags := map[string]string{}

			for key, value := range dpInbound.Tags {
				hasPrefix := func(tagFilter string) bool { return strings.HasPrefix(key, tagFilter) }
				if len(tagFilters) == 0 || slices.ContainsFunc(tagFilters, hasPrefix) {	
					tags[key] = value
				}
			}
```